host/frontend/host-orchestrator/static/js/index.js

```diff
@@ -51,51 +54,103 @@ class DeviceListApp {
     let ul = document.getElementById('device-list');
     ul.innerHTML = '';
     let count = 1;
-    let device_to_button_map = {};
     for (const devId of device_ids) {
-      const buttonId = 'connect_' + count++;
-      let entry = this.#createDeviceEntry(devId, buttonId);
+      const showId = 'connect_' + count++;
+      const launchId = 'connect_' + count++;
+      let entry = this.#createDeviceEntry(devId, showId, launchId);
       ul.appendChild(entry);
-      device_to_button_map[devId] = buttonId;
-    }
-
-    for (const [devId, buttonId] of Object.entries(device_to_button_map)) {
-      let button = document.getElementById(buttonId);
-      button.addEventListener('click', evt => {
-        this.#selectDeviceCb(devId);
+      let launchBtn = document.getElementById(launchId);
+      launchBtn.addEventListener('click', evt => {
+        this.#openDeviceInNewTab(devId);
+      });
+      let showBtn = document.getElementById(showId);
+      showBtn.addEventListener('click', evt => {
+        let shown = this.#toggleDevice(devId);
+        if (shown) {
+          showBtn.textContent = "visibility";
+          showBtn.title = 'Hide device';
+        } else {
+          showBtn.textContent = "visibility_off";
+          showBtn.title = 'Show device';
+        }
       });
     }
   }
 
-  #createDeviceEntry(devId, buttonId) {
+  #createDeviceEntry(devId, showId, launchId) {
```

Comment:
Rather than all of these individual `createElement` calls, is it possible to use a [`<template>` element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/template)?

Reply:
done